core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1752,6 +1755,54 @@ public void testUpgradeMetadataThroughTableProperty(int baseFormatVersion, int n
         .containsExactly(entry("key2", "val2"));
   }
 
+  @ParameterizedTest
+  @EnumSource(
+      value = FileFormat.class,
+      mode = EnumSource.Mode.EXCLUDE,
+      names = {"PARQUET", "PUFFIN", "METADATA"})
+  public void testCreateMetadataVersionWithIncompatibleFileFormat(FileFormat targetFormat) {
+    assertThatThrownBy(
+            () ->
+                TableMetadata.newTableMetadata(
+                    new Schema(),
+                    PartitionSpec.unpartitioned(),
+                    null,
+                    ImmutableMap.of(
+                        TableProperties.FORMAT_VERSION, String.valueOf(3),
+                        TableProperties.DEFAULT_FILE_FORMAT, targetFormat.name())))
+        .hasMessageContaining("file format is not supported in table format version");
+  }
+
+  @ParameterizedTest
+  @MethodSource("upgradeFormatVersionProvider")
+  public void testUpgradeMetadataWithIncompatibleFileFormat(
+      int baseFormatVersion, int newFormatVersion) {
+    TableMetadata meta =
+        TableMetadata.newTableMetadata(
+            new Schema(),
+            PartitionSpec.unpartitioned(),
+            null,
+            ImmutableMap.of(TableProperties.FORMAT_VERSION, String.valueOf(baseFormatVersion)));
+
+    if (newFormatVersion <= 2) {
```

Comment:
I'd rather use `assumeThat(newFormatVersion).isGreaterThan(2)` at the beginning of the test as this is clearer when a test method is skipped and when it isn't

Reply:
Nice, didn't know about this